site/docs/pages/wallet/is-wallet-a-smart-wallet.mdx

```diff
@@ -0,0 +1,37 @@
+# `isWalletASmartWallet`
+
+The `isWalletASmartWallet` utility is designed to verify if a given sender address is a Smart Wallet proxy with the expected implementation before sponsoring a transaction.
+
+## Usage
+
+:::code-group
+
+```tsx [code]
+import { isWalletASmartWallet } from '@coinbase/onchainkit/wallet';
+import type { UserOperation } from 'permissionless';
+import type { PublicClient } from 'viem';
```

Comment:
We are not passing the PublicClient type into `isWalletASmartWallet`. But we are passing the `client` instanciated in the app.